cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -228,16 +238,64 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
     }
 
     public static CruiseControl updateConfiguration(CruiseControlSpec spec, CruiseControl cruiseControl) {
-        CruiseControlConfiguration configuration = new CruiseControlConfiguration(spec.getConfig().entrySet());
-        for (Map.Entry<String, String> entry : configuration.getCruiseControlDefaultPropertiesMap().entrySet()) {
-            if (configuration.getConfigOption(entry.getKey()) == null) {
-                configuration.setConfigOption(entry.getKey(), entry.getValue());
+        CruiseControlConfiguration userConfiguration = new CruiseControlConfiguration(spec.getConfig().entrySet());
+        for (Map.Entry<String, String> defaultEntry : userConfiguration.getCruiseControlDefaultPropertiesMap().entrySet()) {
+            if (userConfiguration.getConfigOption(defaultEntry.getKey()) == null) {
+                userConfiguration.setConfigOption(defaultEntry.getKey(), defaultEntry.getValue());
             }
         }
-        cruiseControl.setConfiguration(configuration);
+        // Ensure that the configured anomaly.detection.goals are a sub-set of the default goals
+        checkGoals(userConfiguration);
+        cruiseControl.setConfiguration(userConfiguration);
         return cruiseControl;
     }
 
+    /**
+     *  This method ensures that the checks in cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java
+     *  sanityCheckGoalNames() method (L118)  don't fail if a user submits custom default goals that have less members then the default
+     *  anomaly.detection.goals.
+     * @param configuration The configuration instance to be checked.
+     * @throws UnsupportedOperationException If the configuration contains self.healing.goals configurations.
+     */
+    public static void checkGoals(CruiseControlConfiguration configuration) {
+        // If self healing goals are defined then these take precedence
+        if (configuration.getConfigOption(CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY) != null) {
+            String selfHealingGoalsString = configuration.getConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY);
+            List<String> selfHealingGoals = Arrays.asList(selfHealingGoalsString.split("\\s*,\\s*"));
+            if (!selfHealingGoals.isEmpty()) {
+                throw new UnsupportedOperationException("Cruise Control's self healing functionality is not currently supported. Please remove " +
+                        CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY + " config");
+            }
+        }
```

Comment:
Could this if-statement block be replaced by putting `self.healing.goals` in the Forbbiden_Options [1] or Forbidden_Prefixes field [2]?


[1] https://github.com/tomncooper/strimzi-kafka-operator/blob/b4c02a4a3aa2d1a65049d21039616f6cd67a10d1/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControlConfiguration.java#L48
[2] https://github.com/strimzi/strimzi-kafka-operator/blob/master/api/src/main/java/io/strimzi/api/kafka/model/CruiseControlSpec.java#L35-L38

Reply:
So I thought about doing that but if we ever support self healing we will need to deal with this issue here so I thought there was no harm in putting the scaffolding in place now.